v2/client.go

```diff
@@ -48,6 +49,9 @@ type HTTPConfig struct {
 
 	// Proxy configures the Proxy function on the HTTP client.
 	Proxy func(req *http.Request) (*url.URL, error)
+
+	// WriteEncoding specifies the encoding of write request, currently support "gzip"
+	WriteEncoding string
```

Comment:
Can you turn this into its own type and an enum?

```
type ContentEncoding string

const (
    DefaultEncoding ContentEncoding = ""
    GzipEncoding ContentEncoding = "gzip"
)
```

This helps allow the type system to point the user to what is available rather than relying on just a comment.

Reply:
done